plugins/maxruntime/maxruntime.go

```diff
@@ -0,0 +1,88 @@
+package maxruntime
+
+import (
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+)
+
+type pluginProvider struct {
+	plugins.PluginProviderBase
+}
+
+type plugin struct {
+	plugins.PluginBase
+}
+
+type taskPlugin struct {
+	plugins.TaskPluginBase
+	maxRunTime int
+	done       chan bool
+	log        *logrus.Entry
+}
+
+type payloadType struct {
+	MaxRunTime int `json:"maxRunTime"`
+}
+
+var payloadSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"maxRunTime": schematypes.Integer{
+			MetaData: schematypes.MetaData{
+				Title:       "Task execution timeout (ms)",
+				Description: "Kill the task if it exceedes the timeout value.",
+			},
+			Minimum: 0,
+			Maximum: 24 * 60 * 60 * 1000,
+		},
+	},
+}
+
+func (pluginProvider) NewPlugin(plugins.PluginOptions) (plugins.Plugin, error) {
+	return plugin{}, nil
+}
+
+func (plugin) PayloadSchema() schematypes.Object {
+	return payloadSchema
+}
+
+func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlugin, error) {
+	var p payloadType
+	err := payloadSchema.Map(options.Payload, &p)
+	if err == schematypes.ErrTypeMismatch {
+		panic("internal error -- type mismatch")
+	} else if err != nil {
+		return nil, engines.ErrContractViolation
+	}
+
+	return &taskPlugin{
+		TaskPluginBase: plugins.TaskPluginBase{},
+		maxRunTime:     p.MaxRunTime,
+		done:           make(chan bool),
+		log:            options.Log,
+	}, nil
+}
+
+func (tp *taskPlugin) Started(sandbox engines.Sandbox) error {
+	go func() {
+		select {
+		case <-time.After(time.Duration(tp.maxRunTime) * time.Millisecond):
+			tp.log.Error("Task expired")
```

Comment:
Log message is not correct...
Please explain what happened here. MaxRunTime was excedded and therefore the task was killed... etc...